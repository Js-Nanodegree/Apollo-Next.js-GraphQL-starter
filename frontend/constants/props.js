import { shape, string } from 'prop-types';

export const user = shape({
  __typename: string.isRequired,
  _id: string.isRequired,
  email: string.isRequired,
  firstName: string.isRequired,
  lastName: string.isRequired
});
