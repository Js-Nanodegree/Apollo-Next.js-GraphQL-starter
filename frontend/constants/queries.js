import gql from 'graphql-tag';

export const CURRENT_USER_QUERY = gql`
  query {
    Me {
      _id
      email
      firstName
      lastName
    }
  }
`;
