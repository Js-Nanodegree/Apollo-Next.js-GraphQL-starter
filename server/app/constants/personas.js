export default [
    {level: 0, label: "Assistant"},
    {level: 0, label: "Junior Manager"},
    {level: 0, label: "Manager"},
    {level: 0, label: "Analyst"},
    {level: 0, label: "Category Manager"},
    {level: 0, label: "Category Director"},
    {level: 0, label: "Senior Manager"},
    {level: 0, label: "Performance"},
    {level: 0, label: "Director"},
    {level: 0, label: "Head of"},
    {level: 0, label: "Head of Excellence"},
    {level: 0, label: "Head of Indirect"},
    {level: 0, label: "Head of Direct"},
    {level: 0, label: "VP"},
    {level: 0, label: "SVP"},
    {level: 0, label: "CPO"},
    {level: 0, label: "Global CPO"},
    {level: 0, label: "C-Level"},
    {level: 0, label: "Consultant"},
    {level: 0, label: "Partner"},
    {level: 0, label: "Lecturer"},
    {level: 0, label: "Student"},
    {level: 0, label: "Journalist"},
    {level: 0, label: "Buyer"}
]