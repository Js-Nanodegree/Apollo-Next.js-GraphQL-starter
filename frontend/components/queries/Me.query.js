import { shape, string, func } from 'prop-types';
import { Query } from 'react-apollo';
import { CURRENT_USER_QUERY } from '../../constants/queries';

export const MeProps = shape({
  __typename: string.isRequired,
  _id: string.isRequired,
  email: string.isRequired,
  firstName: string.isRequired,
  lastName: string.isRequired
});

const Me = props => (
  <Query {...props} query={CURRENT_USER_QUERY}>
    {payload => props.children(payload)}
  </Query>
);

Me.propTypes = {
  children: func.isRequired
};

export default Me;
