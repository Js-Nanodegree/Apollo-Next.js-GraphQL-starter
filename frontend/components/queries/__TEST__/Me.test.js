import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';
import { CURRENT_USER_QUERY } from '../../../constants/queries';
import Me from '../Me.query';

const mocks = [
  {
    request: {
      query: CURRENT_USER_QUERY
    },
    result: {
      __typename: 'User',
      _id: '123456',
      email: 'test@example.com',
      firstName: 'Test',
      lastName: 'Example'
    }
  }
];

describe('Me query component', () => {
  it('renders without error', () => {
    const component = renderer
      .create(
        <MockedProvider mocks={mocks}>
          <Me>{({ data }) => <p>{JSON.stringify(data)}</p>}</Me>
        </MockedProvider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
