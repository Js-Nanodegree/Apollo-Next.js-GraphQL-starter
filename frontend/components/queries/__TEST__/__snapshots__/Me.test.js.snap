// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me query component renders without error 1`] = `
<p>
  {}
</p>
`;
